src/panels/lovelace/cards/hui-grid-card.ts

```diff
@@ -87,10 +87,14 @@ class HuiGridCard extends HuiStackCard<GridCardConfig> {
           grid-column: 1 / 1;
         }
 
-        :host([square]) #root > *:first-child {
+        :host([square]) #root > *:not([invisible]) {
           grid-row: 1 / 1;
           grid-column: 1 / 1;
         }
+        :host([square]) #root > *:not([invisible]) ~ *:not([invisible]) {
```

Comment:
Could also be done with `[style*='display: none']` instead of `[invisible]`

Reply:
`style*` seems to be very picky about exact formatting in the styles. I feel there's an increased risk of mistakes with that.